shoop/core/models/_base.py

```diff
@@ -47,28 +47,37 @@ class Meta:
         abstract = True
 
 
-class ImmutableMixin(object):
+class ChangeProtected(object):
+    protected_fields = None
     unprotected_fields = []
-    immutability_message = _("Cannot change immutable object that is in use")
+    change_protect_message = _("Following fields cannot be changed")
 
     def clean(self, *args, **kwargs):
-        super(ImmutableMixin, self).clean(*args, **kwargs)
+        super(ChangeProtected, self).clean(*args, **kwargs)
         if self.pk:
-            if self._has_any_protected_field_changed() and self._is_in_use():
-                raise ValidationError(self.immutability_message)
+            changed_protected_fields = self._get_changed_protected_fields()
+            if changed_protected_fields and self._is_in_use():
+                message = "{change_protect_message}: {fields}".format(
+                    change_protect_message=self.change_protect_message,
+                    fields=", ".join(sorted(changed_protected_fields)),
+                )
+                raise ValidationError(message)
 
     def save(self, *args, **kwargs):
         self.clean()
-        super(ImmutableMixin, self).save(*args, **kwargs)
+        super(ChangeProtected, self).save(*args, **kwargs)
 
     def _is_in_use(self):
```

Comment:
This PR didn't touch it.

Anyway, something being underscored does not mean it's private. It means that it's non-public. Quoting from PEP8:

> Use one leading underscore only for non-public methods and instance variables.

Being non-public does not mean it's not supposed to be overridden, it just means it's not supposed to be called from outside the class hierarchy. I.e. underscoring hints more about being protected than private (in C++/Java terms). But, yeah this is not really well defined anywhere and people's expectation might vary. Good docstring would help much.